File: c++/algorithms/task24.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
    int maxArea(std::vector<int>& height) {
        
        int mx = 0;
        int left = 0;
        int right = height.size() - 1;

        while (left < right) {  
            int currentHeight = std::min(height[left], height[right]);
            int currentWidth = right - left;
            mx = std::max(currentHeight * currentWidth, mx);

            if (height[left] < height[right]) {
                left++;  
            } else {
                right--;  
            }
        }

        return mx;
    }
};

File: python/task24.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        
        mx = 0

        left = 0
        right = len(height)-1

        c = 0
        while left != right:

            mx = max((min(height[left], height[right]) * (right - left)), mx)
            if height[left] < height[right]:
                left += 1

            else:
                right -= 1



        return mx
